src/components/Interactive.d.ts

```diff
@@ -0,0 +1,12 @@
+import { CSSProperties, PropsWithChildren } from 'react'
+
+
+export type InteractiveProps<Tag extends keyof JSX.IntrinsicElements> = PropsWithChildren<{
+  as: Tag
+  disabled?: boolean
+  hover?: Pick<CSSProperties, 'background' | 'backgroundColor' | 'border' | 'color' | 'boxShadow' | 'opacity' | 'textDecoration'>
+} & JSX.IntrinsicElements[Tag]>
+
+const Interactive: <Tag>(props: InteractiveProps<Tag>) => JSX.Element
```

Comment:
The `JSX.Element` return type does not imply using JSX syntax.

https://www.typescriptlang.org/docs/handbook/jsx.html#function-component